cmd/cstor-pool-mgmt/pool/pool.go

```diff
@@ -0,0 +1,183 @@
+
+	// To generate mirror disk0 disk1 mirror disk2 disk3 format.
+	for i, disk := range cStorPool.Spec.Disks.DiskList {
+		if cStorPool.Spec.PoolSpec.PoolType == "mirror" && i%2 == 0 {
+			createAttr = append(createAttr, "mirror")
+		}
+		createAttr = append(createAttr, disk)
+	}
+	return createAttr
+
+}
+
+// CheckValidPool checks for validity of CStorPool resource.
+func CheckValidPool(cStorPool *apis.CStorPool) error {
+	if string(cStorPool.ObjectMeta.UID) == "" {
+		return fmt.Errorf("Poolname/UID cannot be empty")
+	}
+	if len(cStorPool.Spec.Disks.DiskList) < 1 {
+		return fmt.Errorf("Disk name(s) cannot be empty")
+	}
+	if cStorPool.Spec.PoolSpec.PoolType == "mirror" &&
+		len(cStorPool.Spec.Disks.DiskList)%2 != 0 {
+		return fmt.Errorf("Mirror poolType needs even number of disks")
+	}
+	return nil
+}
+
+// GetPoolName return the pool already created.
+func GetPoolName() ([]string, error) {
+	GetPoolStr := []string{"get", "-Hp", "name", "-o", "name"}
+	poolNameByte, err := RunnerVar.RunStdoutPipe(PoolOperator, GetPoolStr...)
+	if err != nil || string(poolNameByte) == "" {
+		glog.Errorf("Unable to get pool: %v", string(poolNameByte))
+		return []string{}, nil
+	}
+	noisyPoolName := string(poolNameByte)
+	sepNoisyPoolName := strings.Split(noisyPoolName, "\n")
+	var poolNames []string
+	for _, poolName := range sepNoisyPoolName {
+		poolName = strings.TrimSpace(poolName)
+		poolNames = append(poolNames, poolName)
+	}
+	return poolNames, nil
+}
+
+// DeletePool destroys the pool created.
+func DeletePool(poolName string) error {
+	deletePoolStr := []string{"destroy", poolName}
+	stdoutStderr, err := RunnerVar.RunCombinedOutput(PoolOperator, deletePoolStr...)
+	if err != nil {
+		glog.Errorf("Unable to delete pool: %v", string(stdoutStderr))
+		return err
+	}
+	return nil
+}
+
+// SetCachefile is to set the cachefile for pool.
+func SetCachefile(cStorPool *apis.CStorPool) error {
+	poolNameUID := "cstor-" + string(cStorPool.ObjectMeta.UID)
+	setCachefileStr := []string{"set", "cachefile=" + cStorPool.Spec.PoolSpec.CacheFile,
+		poolNameUID}
+	stdoutStderr, err := RunnerVar.RunCombinedOutput(PoolOperator, setCachefileStr...)
+	if err != nil {
+		glog.Errorf("Unable to set cachefile: %v", string(stdoutStderr))
+		return err
+	}
+	return nil
+}
+
+// CheckForZrepl is blocking call for checking status of zrepl in cstor-pool container.
+func CheckForZrepl() {
+	for {
+		_, err := RunnerVar.RunCombinedOutput(PoolOperator, "status")
+		if err != nil {
+			time.Sleep(ZreplRetryInterval)
+			glog.Infof("Waiting for zpool replication container to start......")
+			continue
+		}
+		break
+	}
+}
+
+// LabelClear is to clear zpool label on disks.
+func LabelClear(disks []string) error {
+	for _, disk := range disks {
+		labelClearStr := []string{"labelclear", "-f", disk}
+		stdoutStderr, err := RunnerVar.RunCombinedOutput(PoolOperator, labelClearStr...)
+		if err != nil {
+			glog.Errorf("Unable to clear label: %v", string(stdoutStderr))
+			return err
```

Comment:
Should not error out if the operation fails for one disk. Instead, capture the error and proceed with clearing the labels on other disks.

Reply:
will change